src/dawg.cpp

```diff
@@ -15,168 +15,171 @@
-		DAWG_ERROR("bad configuration");
-		return EXIT_FAILURE;
-	}
-	// create sets of aligned sequences;
-	dawg::alignment aln;
-	kimura.pre_walk(aln);
-	for(unsigned int i=0;i<num_reps;++i) {
-		kimura.walk(aln);
-		write_aln(aln);
-	}	
-	return EXIT_SUCCESS;
+    // std::string _temp(" I love %r/%R/%%/%.  Do you?"), _out;
+    //_out = boost::algorithm::replace_all_regex_copy(_temp,
+    // boost::regex("%(r)|%(R)|%(%)"), 	std::string("?1x:?2y:z"), match_default
+    // | format_all); cout << _out << endl << endl;
+
+    if(arg.version) {
+        cerr << endl << VERSION_MSG << endl << endl;
+        return EXIT_SUCCESS;
+    }
+    if(arg.help_trick) {
+        cerr << endl << VERSION_MSG << endl << endl;
+        ma::help(cerr);
+        return EXIT_SUCCESS;
+    }
+    if(arg.help || arg.input.empty()) {
+        cerr << endl << VERSION_MSG << endl << endl;
+        cerr << "Usage:\n  " << runname
+             << " [options] trick-1.dawg trick-2.dawg ..." << endl
+             << endl;
+        cerr << desc << endl;
+        return EXIT_SUCCESS;
+    }
+
+    // if(arg.quiet)
+    //	cerr.clear(ios::failbit);
+    trick input;
+
+    bool ret = true;
+    for(string &ss : arg.input) {
+        ret &= trick::parse_file(input, ss.c_str());
+    }
+
+    if(!ret) return EXIT_FAILURE;
+    // process aliases
+    input.read_aliases();
+
+    global_options glopts;
+    glopts.read_section(input.data.front());
+
+    unsigned int num_reps = (arg.reps > 0) ? arg.reps : glopts.sim_reps;
+
+    dawg::output write_aln;
+    const char *file_name =
+        arg.output.empty() ? glopts.output_file.c_str() : arg.output.c_str();
+    // bool split  = (!vm["split"].defaulted()) ? arg.split :
+    // glopts.output_split; bool append = (!vm["append"].defaulted()) ?
+    // arg.append : glopts.output_append;
+
+    bool split = arg.split.get_value_or(glopts.output_split);
+    bool append = arg.append.get_value_or(glopts.output_append);
+    bool label = arg.label.get_value_or(glopts.output_label);
+
+    if(!write_aln.open(file_name, num_reps - 1, split, append, label)) {
+        DAWG_ERROR("bad configuration");
+        return EXIT_FAILURE;
+    }
+    write_aln.set_blocks(
+        glopts.output_block_head.c_str(), glopts.output_block_between.c_str(),
+        glopts.output_block_tail.c_str(), glopts.output_block_before.c_str(),
+        glopts.output_block_after.c_str());
+
+    vector<dawg::ma> configs;
+    if(!dawg::ma::from_trick(input, configs)) {
+        DAWG_ERROR("bad configuration");
+        return EXIT_FAILURE;
+    }
+
+    // Create the object that will do all the simulation
+    // work for us.  Configure its sections.
+    dawg::matic simulation;
+    // if a seed was specified, use it
+    if(arg.seed != 0) {
+        simulation.seed(arg.seed);
+    } else if(!glopts.sim_seed.empty()) {
+        simulation.seed(glopts.sim_seed.begin(), glopts.sim_seed.end());
+    } else {
+        simulation.auto_seed_seq();
```

Comment:
Changed the variable name to simulation and added the else statement to pick a random seed if no user specified ones.